app/controllers/sign_in_kids_controller.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class SignInKidsController < ApplicationController
+  layout 'initial_application'
+
+  def new; end
+
+  def create
+    account = Account.find_by(parental_key)
```

Comment:
WDYT if we do this:
1. find "account share" by the parental key
2. we can get the "account" from this share

As a result, we do not need another parental key inside the "account" model.